File: cpp/lv0/잘라서_배열로_저장하기/잘라서_배열로_저장하기.cpp
#include <string>
#include <vector>

using namespace std;

// 1. 반복문을 사용하여 my_str을 n만큼 잘라서 answer에 넣는다.

vector<string> solution(string my_str, int n) {
    vector<string> answer;
    string temp = my_str;
    while (true) {
        answer.push_back(temp.substr(0, n));
        temp = temp.substr(n);
        
        // 남은 길이가 0일때 push_back을 하면 ""을 집어넣는다. 그래서 그대로 리턴
        if (temp.length() == 0) {
            return answer;
        // 남은 길이가 n보다 작으면 그대로 다 넣고 리턴
        } else if (temp.length() < n) {
            answer.push_back(temp);
            return answer;
        }
    }
    return answer;
}

File: go/lv0/잘라서_배열로_저장하기/잘라서_배열로_저장하기.go
func solution(my_str string, n int) []string {
    if len(my_str) < n {
        return []string{my_str}
    }
    // 정답을 담을 슬라이스
    var result []string
    // temp에 my_str을 복사
    temp := my_str
    for {
        // 예외처리 -> 만약 여기서 if와 else if의 내용이 바뀌면
        // 0도 < n 이기 때문에 result에 할당하고, 에러가 난다.
        if len(temp) == 0 {
            return result
        } else if len(temp) < n {
            result = append(result, temp)
            return result
        } 
        // 위에서 걸리지 않을 경우 result에 temp의 앞에서 n만큼을 담아줌
        result = append(result, temp[:n])
        temp = temp[n:]
        
    }
}